common/flatpak-run.c

```diff
@@ -1622,13 +1622,31 @@ job_removed_cb (SystemdManager *manager,
     g_main_loop_quit (data->main_loop);
 }
 
+static gchar *
+systemd_unit_name_escape (const gchar *in) {
+  /* Adapted from systemd source */
+  gchar * const ret = g_new (gchar, strlen (in) * 4 + 1);
+  gchar *out = ret;
+
+  for (; *in; in++) {
+    if (g_ascii_isalnum (*in) || *in == ':' || *in == '_' || *in == '.')
```

Comment:
Flatpak appids are limited to `a-z0-9-_.`, so this will really only affect `-`. I guess its nice to have the exact same escape code as in systemd though,